### DISCUSSION TITLE
Custom action - Enclosure Plugin Toggle GPIO

### DISCUSSION THREAD

**Discussion by tjharvey71** [UnchartedBull/OctoDash#2456] [Q&A] [answered]
Hi I have a led light bar controlled via a GPIO controlled relay and the enclosure plugin.
I can turn the light bar on/off manually and automatically in octoprint.

I tried to create custom action button in octodash but when I select it the light does not turn on. 
The light bar is ID 1 in the enclosure plugin and GPIO 24. I must be missing something obvious.
snip from the json file:
{
                                        "color": "#e1b12c",
                                        "command": "[!OUTPUT]1,on",
                                        "confirm": false,
                                        "exit": false,
                                        "icon": "lightbulb"
                                },
thanks
TJH

**Comment by tjharvey71** [comment]
for now I added 2 custom actions using the enclosure plugin webui:
 {
                                        "color": "#e1b12c",
                                        "command": "[!WEB]http://xxx.x.x.x/plugin/enclosure/setIO?status=true&index_id=1&apikey=xxxxxxxxx",
                                        "confirm": false,
                                        "exit": false,
                                        "icon": "lightbulb"
                                },
                                {
                                        "color": "#e84118",
                                        "command": "[!WEB]http://xxx.x.x.x/plugin/enclosure/setIO?status=false&index_id=1&apikey=xxxxxxxxx",
                                        "confirm": false,
                                        "exit": false,
                                        "icon": "lightbulb"
                                },

Each click goes to a "loading website page" that you have to "x" to close.

thanks,
TJH

**Reply by Ruhel786** [reply]
Hi Im trying to do the same thing with no luck. Did you get this working. The [!OUTPUT] does nothing for me.

**Comment by UnchartedBull** [comment, marked as answer]
[!OUTPUT] will be added with the next release, so please wait for that :)

**Comment by marc965** [comment]
Hi!
tried to power on Led with "command": "[!OUTPUT]1,on" but not working....
tried fruther to turn on fan with "command": "[!OUTPUT_PWM]3,50" but not working too.

Any informations for me when the next release is planned?
